SpotifySearch/SpotifySearch/SpotifyTableViewController.swift

```diff
@@ -0,0 +1,129 @@
+//
+//  SpotifyTableViewController.swift
+//  SpotifySearch
+//
+//  Created by Annie Tung on 10/28/16.
+//  Copyright © 2016 Annie Tung. All rights reserved.
+//
+
+import UIKit
+
+class SpotifyTableViewController: UITableViewController {
+    
+    var kaskade = [Kaskade]()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        APIRequestManager.manager.getSpotifyData(apiEndPoint: "https://api.spotify.com/v1/search?q=automatic&type=album&limit=50") { (data: Data?) in
+                if let k = self.validKaskadeData(data: data!) {
+                    self.kaskade = k
+                    print("*****kaskade data has been populated")
+            }
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+            }
+        }
+    }
+    
+    func validKaskadeData (data: Data) -> [Kaskade]? {
+        var k = [Kaskade]()
+        
+        do{
+            let kaskadeData: Any = try JSONSerialization.jsonObject(with: data, options: [])
+            
+            guard let kaskadeCasted: [String:Any] = kaskadeData as? [String:Any] else {
+                print("*****There was an error casting from String:any at \(kaskadeData)")
+                return nil
+            }
+            print("*****Data was created \(kaskadeData)")
+            
+            guard let kaskadeArray: [String:Any] = kaskadeCasted["albums"] as? [String:Any] else {
+                print("*****There was an error casting from [[String:any]] at \(kaskadeCasted)")
+                return nil
+            }
+            print("*****Array was created \(kaskadeCasted)")
+            
+            guard let kaskadeItems: [[String:Any]] = kaskadeArray["items"] as? [[String: Any]] else {
+                print("*****There was an error casting from String:any at \(kaskadeArray)")
+                return nil
+            }
+            print("*****Valid array was created \(kaskadeItems)")
+            
+            kaskadeItems.forEach({ kaskadeObject in
+                
+                guard let albumName: String = kaskadeObject["name"] as? String,
```

Comment:
You were so close. You weren't parsing correctly which is why your tableview wasn't loading. See below. I think you mis-cast some variable as Dictionaries as opposed to Array of Dictionaries. 

  
items.forEach {
                guard let albumName = $0["name"] as? String else { return }
                // pull artist
                guard let artistList = $0["artists"] as? [[String: Any]] else { return }
                guard let mainArtist = artistList[0]["name"] as? String else { return } 
                // pull image
                guard let images = $0["images"] as? [[String: Any]] else { return }
                guard let thumbnail = images[2]["url"] as? String else { return }
                guard let largeImage = images[1]["url"] as? String else { return }
                let albumObject = SpotifyAlbum(albumName: albumName, artistName: mainArtist, albumImage: AlbumImage(thumbnailImage: thumbnail, fullImage: largeImage))
                allAlbums.append(albumObject)
            }